helper/resource/state.go

```diff
@@ -213,7 +216,9 @@ func (conf *StateChangeConf) WaitForState() (interface{}, error) {
 
 			// still waiting, store the last result
 			lastResult = r
-
+		case <-ctx.Done():
+			// this is a simple approach but swallows lastResult.Error
+			return lastResult.Result, ctx.Err()
```

Comment:
considering `ctx.Err` is non-nil here, result should be `nil`

Reply:
Yeah wasn't sure if we should return what we got at the very least? Would it not be beneficial to give the developer options? Not a strong opinion happy to keep it simple with `nil` if that's a better practice.